src/evict/evict_lru.c

```diff
@@ -2898,28 +2898,23 @@ __wti_evict_app_assist_worker(WT_SESSION_IMPL *session, bool busy, bool readonly
         if (!__wt_evict_needed(session, busy, readonly, &pct_full) ||
           (pct_full < 100.0 &&
             (__wt_atomic_loadv64(&evict->eviction_progress) > initial_progress + max_progress)))
-            goto err;
+            break;
 
         if (!__evict_check_user_ok_with_eviction(session, busy))
-            /* At this point ret can only be 0, so it's a clean exit from the loop. */
-            goto err;
+            break;
 
         /* Evict a page. */
-        switch (ret = __evict_page(session, false)) {
-        case 0:
+        ret = __evict_page(session, false);
+        if (ret == 0) {
+            /* If the caller holds resources, we can stop after a successful eviction. */
             if (busy)
-                goto err;
-        /* FALLTHROUGH */
-        case EBUSY:
-            break; /* Continue the loop. */
-        case WT_NOTFOUND:
+                break;
+        } else if (ret == WT_NOTFOUND) {
             /* Allow the queue to re-populate before retrying. */
             __wt_cond_wait(session, conn->evict_threads.wait_cond, 10 * WT_THOUSAND, NULL);
             evict->app_waits++;
-            break; /* Continue the loop. */
-        default:
-            goto err;
-        }
+        } else if (ret != EBUSY)
+            WT_ERR(ret);
```

Comment:
Nit: Would using a `goto err` here be less confusing, given that `ret` can’t be `0` at this point?

Reply:
The use of `WT_ERR(ret)` even in cases where ret is known to be non-zero is the standard, common usage in WT.

Reply:
Thanks @ershov. As @sueloverso says, `WT_ERR` is standard usage here. And consistency has value. 

Also using `WT_ERR` is safer if the code changes. Maybe some future change would allow `ret == 0` here (although it's hard to imagine). Or maybe we will add some sort of error tracking/debugging functionality inside `WT_ERR`, etc.